auto-value-parcel/src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java

```diff
@@ -214,6 +215,7 @@ FieldSpec generateCreator(ProcessingEnvironment env, List<Property> properties,
 
     MethodSpec.Builder createFromParcel = MethodSpec.methodBuilder("createFromParcel")
         .addAnnotation(Override.class)
+        .addAnnotation(createSuppressUncheckedWarningAnnotation())
```

Comment:
Yes. This library bears the burden of that and will have test cases accordingly.

Suppressing warnings where there are none is deemed an incorrect practice by most. And by unconditionally suppressing you are equally as likely to introduce a type system bug that goes uncaught until runtime, no?